internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -38,12 +40,39 @@ func (f *FirstReadyCluster) FindCluster(kafka *dbapi.KafkaRequest) (*api.Cluster
 		SupportedInstanceType: kafka.InstanceType,
 	}
 
-	cluster, err := f.ClusterService.FindCluster(criteria)
-	if err != nil {
-		return nil, err
+	clusters, findAllClusterErr := f.ClusterService.FindAllClusters(criteria)
+	if findAllClusterErr != nil {
+		return nil, findAllClusterErr
+	}
+
+	// Get total number of streaming unit used per cluster that matches with criteria
+	streamingUnitCountPerRegionList, countStreamingUnitErr := f.KafkaService.CountStreamingUnitByRegionAndInstanceType()
+	if countStreamingUnitErr != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, countStreamingUnitErr, "failed to get count of streaming units by region and instance type")
+	}
+
+	streamingUnitsUsed := make(map[string]int32, len(clusters))
+
+	for _, cluster := range clusters {
+		streamingUnits := streamingUnitCountPerRegionList.GetStreamingUnitCountForClusterAndInstanceType(cluster.ClusterID, kafka.InstanceType)
+		streamingUnitsUsed[cluster.ClusterID] = int32(streamingUnits)
 	}
 
-	return cluster, nil
+	// Find first ready cluster that has remaining capacity (streaming unit count is less than the cluster maxUnit)
+	for _, cluster := range clusters {
+		capacityInfo, getCapacityInfoErr := cluster.RetrieveDynamicCapacityInfo()
+		if getCapacityInfoErr != nil {
+			return nil, errors.NewWithCause(errors.ErrorGeneral, getCapacityInfoErr, "failed to retrieve capacity information for cluster %s", cluster.ClusterID)
+		}
+		maxStreamingUnits := capacityInfo[kafka.InstanceType].MaxUnits
+
+		if streamingUnitsUsed[cluster.ClusterID] < maxStreamingUnits {
+			return cluster, nil
+		}
```

Comment:
Let's also include the size of the kafka as well. 
```go
		if (streamingUnitsUsed[cluster.ClusterID] + capacityConsumedOfTheNewKafka)  < maxStreamingUnits {
			return cluster, nil
		}
```
```suggestion
		if streamingUnitsUsed[cluster.ClusterID] < maxStreamingUnits {
			return cluster, nil
		}
```

Reply:
Something like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/fea39a658464424833b7742a13f4320b0f213537/internal/kafka/internal/services/cluster_placement_strategy.go#L131